src/modules/mavlink/mavlink_receiver.h

```diff
@@ -206,74 +206,74 @@ class MavlinkReceiver
-	orb_advert_t _actuator_controls_pubs[4];
-	orb_advert_t _att_sp_pub;
-	orb_advert_t _rates_sp_pub;
-	orb_advert_t _pos_sp_triplet_pub;
-	orb_advert_t _mocap_odometry_pub;
-	orb_advert_t _visual_odometry_pub;
-	orb_advert_t _radio_status_pub;
-	orb_advert_t _ping_pub;
-	orb_advert_t _rc_pub;
-	orb_advert_t _manual_pub;
-	orb_advert_t _obstacle_distance_pub;
-	orb_advert_t _trajectory_waypoint_pub;
-	orb_advert_t _land_detector_pub;
-	orb_advert_t _follow_target_pub;
-	orb_advert_t _landing_target_pose_pub;
-	orb_advert_t _transponder_report_pub;
-	orb_advert_t _collision_report_pub;
-	orb_advert_t _debug_key_value_pub;
-	orb_advert_t _debug_value_pub;
-	orb_advert_t _debug_vect_pub;
-	orb_advert_t _debug_array_pub;
-	orb_advert_t _gps_inject_data_pub;
-	orb_advert_t _command_ack_pub;
+	struct vehicle_attitude_s _att {};
+	struct vehicle_local_position_s _hil_local_pos {};
+	struct vehicle_land_detected_s _hil_land_detector {};
+	struct vehicle_control_mode_s _control_mode {};
+
+	orb_advert_t _accel_pub{nullptr};
+	orb_advert_t _actuator_controls_pubs[4] {nullptr, nullptr, nullptr, nullptr};
+	orb_advert_t _airspeed_pub{nullptr};
+	orb_advert_t _att_sp_pub{nullptr};
+	orb_advert_t _attitude_pub{nullptr};
+	orb_advert_t _baro_pub{nullptr};
+	orb_advert_t _battery_pub{nullptr};
+	orb_advert_t _cmd_pub{nullptr};
+	orb_advert_t _collision_report_pub{nullptr};
+	orb_advert_t _command_ack_pub{nullptr};
+	orb_advert_t _debug_array_pub{nullptr};
+	orb_advert_t _debug_key_value_pub{nullptr};
+	orb_advert_t _debug_value_pub{nullptr};
+	orb_advert_t _debug_vect_pub{nullptr};
+	orb_advert_t _distance_sensor_pub{nullptr};
+	orb_advert_t _flow_distance_sensor_pub{nullptr};
+	orb_advert_t _flow_pub{nullptr};
+	orb_advert_t _follow_target_pub{nullptr};
+	orb_advert_t _global_pos_pub{nullptr};
+	orb_advert_t _gps_inject_data_pub{nullptr};
+	orb_advert_t _gps_pub{nullptr};
+	orb_advert_t _gyro_pub{nullptr};
+	orb_advert_t _hil_distance_sensor_pub{nullptr};
+	orb_advert_t _land_detector_pub{nullptr};
+	orb_advert_t _landing_target_pose_pub{nullptr};
+	orb_advert_t _local_pos_pub{nullptr};
+	orb_advert_t _mag_pub{nullptr};
+	orb_advert_t _manual_pub{nullptr};
+	orb_advert_t _mocap_odometry_pub{nullptr};
+	orb_advert_t _obstacle_distance_pub{nullptr};
+	orb_advert_t _offboard_control_mode_pub{nullptr};
+	orb_advert_t _ping_pub{nullptr};
+	orb_advert_t _pos_sp_triplet_pub{nullptr};
+	orb_advert_t _radio_status_pub{nullptr};
+	orb_advert_t _rates_sp_pub{nullptr};
+	orb_advert_t _rc_pub{nullptr};
+	orb_advert_t _trajectory_waypoint_pub{nullptr};
+	orb_advert_t _transponder_report_pub{nullptr};
+	orb_advert_t _visual_odometry_pub{nullptr};
 
 	static const int _gps_inject_data_queue_size = 6;
 
 	int _actuator_armed_sub{orb_subscribe(ORB_ID(actuator_armed))};
 	int _control_mode_sub{orb_subscribe(ORB_ID(vehicle_control_mode))};
 	int _vehicle_attitude_sub{orb_subscribe(ORB_ID(vehicle_attitude))};
 
-	int _hil_frames;
+	int _hil_frames{0};
+	int _orb_class_instance{-1};
```

Comment:
We probably don't need this at all. Most (need to check) of the mavlink receiver publishes should be using `orb_publish_auto`.

Reply:
Cool. You're correct, `_hil_frames` and `_old_timestamp` only get used here: https://github.com/PX4/Firmware/blob/447e0be2509b1cbd8337c259973d049e37a81488/src/modules/mavlink/mavlink_receiver.cpp#L2137.  I'll deprecate that block in #11274 .